eclass/user-info.eclass

```diff
@@ -37,18 +37,27 @@ egetent() {
 		# lookup by uid/gid
 		local opts
 		if [[ ${key} == [[:digit:]]* ]] ; then
-			[[ ${db} == "user" ]] && opts="-u" || opts="-g"
+			[[ ${db} == "user" ]] && opts=( -u ) || opts=( -g )
 		fi
 
+		# Handle different ROOT
+		[[ -n ${ROOT} ]] && opts+=( -R "${ROOT}" )
+
 		pw show ${db} ${opts} "${key}" -q
 		;;
 	*-openbsd*)
-		grep "${key}:\*:" /etc/${db}
+		grep "${key}:\*:" "${EROOT}/etc/${db}"
 		;;
 	*)
-		# ignore nscd output if we're not running as root
-		type -p nscd >/dev/null && nscd -i "${db}" 2>/dev/null
-		getent "${db}" "${key}"
+		# getent does not support -R option, if we are working on a different
+		# ROOT than /, fallback to grep technique.
+		if [[ -z ${ROOT} ]]; then
+			# ignore nscd output if we're not running as root
+			type -p nscd >/dev/null && nscd -i "${db}" 2>/dev/null
+			getent "${db}" "${key}"
+		else
+			grep "^${key}:" "${EROOT}/etc/${db}"
```

Comment:
If it is a UID, I can do something like this:
```sh
grep "^${key}:" "${EROOT}/etc/${db}" || \
  grep "^*:x:${key}:" "${EROOT}/etc/${db}"
```
It should fail if the key isn't found because it is a UID and so, try again by searching by UID this time.

Reply:
Could do if [[ ... =~ [[:digit:]* ]] like above for the freebsd

I don't think the x is guaranteed though, you could use awk to extract the uid like in that patch in the bug tracker

Reply:
```sh
if [[ ${key} =~ ^[[:digit:]]+$ ]]; then
	grep -E "^([^:]*:){2}${key}" "${ROOT}/etc/${db}"
else
	grep "^${key}:" "${ROOT}/etc/${db}"
fi
```

This should do the trick.

Reply:
btw: `[[ ${key} =~ [[:digit:]]*  ]]` will watch if the key contain a digit which is probably not what we want.

Reply:
You're right, I meant like freebsd is doing [[ ${key} == [[:digit:]]* ]]